src/viadot/sources/eurostat.py

```diff
@@ -0,0 +1,356 @@
+                index = str(
+                    col_index + row_index * len(signal_lists[1].signal_label_list)
+                )
+                if index in eurostat_dictionary["value"].keys():
+                    index_list.append(index)
+                    col_signal_temp.append(col_label)
+                    row_signal_temp.append(row_label)
+
+        indicator_list = [eurostat_dictionary["value"][i] for i in index_list]
+        df.indicator = indicator_list
+        df[signal_lists[1].signal_name] = col_signal_temp
+        df[signal_lists[0].signal_name] = row_signal_temp
+
+        return df
+
+    @add_viadot_metadata_columns
+    def to_df(
+        self,
+        if_empty="warn"
+    ) -> pd.DataFrame:
+        """Function responsible for getting response and creating DataFrame.
+
+        It is using method 'eurostat_dictionary_to_df' with validation
+        of provided parameters and their codes if needed.
+
+        Raises:
+            TypeError: If self.params is different type than a dictionary.
+            TypeError: If self.columns is different type than a list containing strings.
+            APIError: If there is an error with the API request.
+
+        Returns:
+            pd.DataFrame: Pandas DataFrame.
+        """
+        # Checking if params and columns have correct type
+        if not isinstance(self.params, dict) and self.params is not None:
+            raise TypeError("Params should be a dictionary.")
+
+        if not isinstance(self.columns, list) and self.columns is not None:
+            raise TypeError("Requested columns should be provided as list of strings.")
+
+        # Creating url for connection with API
+        url = f"{self.base_url}{self.dataset_code}?format=JSON&lang=EN"
+
+        # Making parameters validation
+        if self.params is not None:
+            self.validate_params(dataset_code=self.dataset_code,
+                                 url=url,
+                                 params=self.params)
+
+        # Getting response from API
+        try:
+            response = handle_api_response(url, params=self.params)
+            data = response.json()
+            data_frame = self.eurostat_dictionary_to_df(["geo", "time"], data)
+        except APIError:
+            self.validate_params(dataset_code=self.dataset_code,
+                                 url=url,
+                                 params=self.params)
+
+        # Merge data_frame with label and last updated date
+        label_col = pd.Series(str(data["label"]), index=data_frame.index, name="label")
+        last_updated__col = pd.Series(
+            str(data["updated"]),
+            index=data_frame.index,
+            name="updated",
+        )
+        data_frame = pd.concat([data_frame, label_col, last_updated__col], axis=1)
+
+        # Validation and transformation of requested column
+        if self.columns is not None:
+            filter_df_columns(data_frame=data_frame, columns=self.columns)
```

Comment:
This function doesn't exist, but if it did, it shouldn't modify input but return a new df:

```suggestion
            df_filtered = filter_df_columns(data_frame=data_frame, columns=self.columns)
```

Reply:
As agreed with Fabio we are not changing any logic at this point. All logic changes going to be issues - so after migration somebody can improve connectors.

Reply:
Sure, pls link the issue here so we can close this

Reply:
https://github.com/dyvenia/viadot/issues/1046